internal/resources/credential/datasource_credential.go

```diff
@@ -39,14 +44,59 @@ func dataSourceCredentialRead(ctx context.Context, d *schema.ResourceData, m int
 
 	name, _ := d.Get(NameKey).(string)
 
-	resp, err = config.TMCConnection.CredentialResourceService.CredentialResourceServiceGet(constructFullname(d))
-	if err != nil || resp == nil {
-		if clienterrors.IsNotFoundError(err) && !helper.IsDataRead(ctx) {
-			_ = schema.RemoveFromState(d, m)
-			return diags
+	getCredentialResourceRetrybleFn := func() (retry bool, err error) {
```

Comment:
nit: Typo getCredentialResourceRetrybleFn  => getCredentialResourceRetryableFunc

Reply:
Fixed. Fyi-  I was just following the same naming conventioneer followed in existing resources with wait timeout